miden-crypto/src/dsa/rpo_falcon512/keys/public_key.rs

```diff
@@ -23,63 +15,38 @@ use crate::{Word, dsa::rpo_falcon512::FALCON_ENCODING_BITS};
 // PUBLIC KEY
 // ================================================================================================
 
-/// A public key for verifying signatures.
-///
-/// The public key is a [Word] (i.e., 4 field elements) that is the hash of the coefficients of
-/// the polynomial representing the raw bytes of the expanded public key. The hash is computed
-/// using Rpo256.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct PublicKey(Word);
+/// Public key represented as a polynomial with coefficients over the Falcon prime field.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PublicKey(pub Polynomial<FalconFelt>);
 
 impl PublicKey {
-    /// Returns a new [PublicKey] which is a commitment to the provided expanded public key.
-    pub fn new(pub_key: Word) -> Self {
-        Self(pub_key)
-    }
-
     /// Verifies the provided signature against provided message and this public key.
     pub fn verify(&self, message: Word, signature: &Signature) -> bool {
-        signature.verify(message, self.0)
+        signature.verify(message, self)
     }
-}
 
-impl From<PubKeyPoly> for PublicKey {
-    fn from(pk_poly: PubKeyPoly) -> Self {
-        let pk_felts: Polynomial<Felt> = pk_poly.0.into();
-        let pk_digest = Rpo256::hash_elements(&pk_felts.coefficients);
-        Self(pk_digest)
+    /// Returns the digest of the public key serving as a commitment to it.
+    pub fn to_word(&self) -> Word {
```

Comment:
I would probably rename this to `to_commitment()`.

Also, it may make sense to implement `SequentialCommit` for `PubKey`.

Reply:
Implemented `SequentialCommit`